File: Chapter-05-Linked-Lists/SList-Remove-Front/SList-Remove-Front.js
// Given a pointer to the first node in the list, remove the head node and return the new head node

// import the list class and node class
var path = require('path'),
    SList= require(path.join(__dirname, '../singleList'))

// Create a new list and populate it
var newSList = new SList()
for(let i = 0; i < 10; i++){
  newSList.add(Math.floor(Math.random() * 100))
}

// print the existing list
console.log(`The existing list is ${newSList.show()} - length is ${newSList.length()}`)

// function to remove the first element of the list
function removeFront(list, head){
  let temp = head
  head = head.next
  list.setHeadNode(head)
  list.recalculateLength()
  return list.getHeadNode()
}

// print the results of the remove function
console.log(removeFront(newSList, newSList.getHeadNode()))
console.log(`The modified list is ${newSList.show()} - length is ${newSList.length()}`)

File: Chapter-05-Linked-Lists/SList-Remove-Front/SList-Remove-Front.py
# Given a pointer to the first node in the list, remove the head node and return the new head node

# import the list class.
import sys
from os.path import abspath, dirname

mainDir = dirname(dirname(abspath(__file__)))
sys.path.append(mainDir)

from singleList import List

# import random module
from random import randrange

# create a new list and populate it
newSList = List()
for i in range(0, 10):
    newSList.add(randrange(100))

# print the existing list
print('The existing list is {} - length is {}').format(newSList.show(), newSList.length())

# function to remove the first element in the list
def removeFront(list, head):
    temp = head
    head = head.next
    list.setHeadNode(head)
    list.recalculateLength()
    return list.getHeadNode()

# print the results of the remove function
print(removeFront(newSList, newSList.getHeadNode()))
print('The modified list is {} - length is {}').format(newSList.show(), newSList.length())
